yunwoo/programmers-1/푸드-파이트-대회.js

```diff
@@ -0,0 +1,26 @@
+function solution(food) {
+  const result = [0];
+
+  for (let i = food.length - 1; i > 0; i--) {
+    if (food[i] % 2 === 0) {
+      for (let j = 1; j < (food[i] + 1) / 2; j++) {
+        result.push(i);
+        result.unshift(i);
+      }
+    } else {
+      for (let j = 1; j < food[i] / 2; j++) {
+        result.push(i);
+        result.unshift(i);
+      }
+    }
+  }
+  return result.join("");
+}
```

Comment:
```suggestion
function solution(food) {
  const result = [0];
  const foodCounts = food.map((el) => el - (el % 2));

  for (let i = foodCounts.length - 1; i > 0; i--) {
      for (let j = 1; j < (foodCounts[i] + 1) / 2; j++) {
        result.push(i);
        result.unshift(i);
      }
  }
    
  return result.join("");
}
```
- 풀이 잘봤습니다 위와 같이 foodCounts를 미리 만들어준 후에 이를 순회하면 for문 내의 분기처리를 굳이 할 필요가 없어집니다 ~ 🙇‍♂️
- 아래 풀이는 로직 자체는 비슷한 맥락인데 push, unshift를 활용하는게 아닌 reduce와, 전개 연산자를 활용한 풀이입니다. 
```js
function solution(food) {
  const foodCounts = food
    .slice(1)
    .reverse()
    .map((el) => el - (el % 2));

  const result = foodCounts.reduce(
    (acc, cur, i) => {
      const numberOfFood = foodCounts.length - i;

      while (cur) {
        acc = [numberOfFood, ...acc, numberOfFood];
        cur -= 2;
      }

      return acc;
    },
    [0]
  );

  return result.join("");
}
```

Reply:
- 와 그런 방법이 있었군요!! 감사합니다 반복구문을 어떻게 없앨까 고민했는데👍
- 하연님 코드 살펴봤는데 for문을 한개로 만드셨더라구요.. 코드 뜯어보겠습니다 🙇‍♂️
- 참고풀이도 살펴보겠습니다 감사합니다 🙇‍♂️👍